src/Components/Generator/DynamicInputContainer.js

```diff
@@ -216,13 +216,34 @@ export default class DynamicInputContainer extends preact.Component {
     }
 
     removeDynamicInput(event) {
-        if (window.confirm(t('confirm-input-remove', 'generator'))) {
-            this.setState(prev => {
-                delete prev.fields[event.target.getAttribute('rel')];
-                return prev;
-            });
-            this.pushStateUp();
+        // prompt only if respective field has value entered while removing
+        if (!this.checkInputData(event)) {
+            this.confirmStateChange(event);
+        } else if(window.confirm(t('confirm-input-remove', 'generator'))) {
+            this.confirmStateChange(event);
+        }   
+    }
+
+    confirmStateChange(event) {
```

Comment:
I think this function name is misleading. Since this might delete data the naming should be very clear.